### DISCUSSION TITLE
Easiest way to import edge list data from a set of parquet files

### DISCUSSION THREAD

**Discussion by Ishitori** [memgraph/memgraph#824] [Q&A] [answered]
I have a set of files in parquet format in S3 which stores edges of my graph. The format is src_id, dst_id, weight. A few questions regarding this

1. What is the easiest way to import this data to Memgraph without converting it to CSV and without sharing S3 credentials? Is there an example for that?
2.  Does Memgraph support truly large graphs of ~20B edges?

**Comment by antejavor** [comment]
Hi @Ishitori, 

1. Currently, we support this (public key auth) via **[GQLAlchemy](https://memgraph.com/docs/gqlalchemy/how-to-guides/table-to-graph-importer) (Python + PyArrow)**.   At the moment, we do not have a native integration (C++) for loading parquet files from S3 as we have for loading CSV files.  This brings me to a potential issue for your use case, we didn't test GQLAlchemy import on that big graph. But feel free to try it out. 

2. Memgraph supports truly large graphs ~20B edges. The question at hand is, do you have enough RAM to store everything natively on a single machine? 
If that is not the case, we are building interesting stuff around distributed and disc storage, it will be out in a few months, if you are interested in a sneak peek, we could arrange a call. 

If you have extra questions, feel free to ask. We have an active discord community. Feel free to join: [](https://discord.gg/memgraph)

**Comment by Ishitori** [comment]
Thanks @antejavor, will try to import the data. Will checkout the discord as well.

**Comment by karrtikiyer** [comment, marked as answer]
@antejavor & @Ishitori : Coming to the original question, is there a way to *directly* import edge list into mem-graph, the answer seems to be *NO*. It has to be converted to a table to graph importer format as described [here](https://memgraph.com/docs/gqlalchemy/how-to-guides/table-to-graph-importer). Is that a fair understanding?

**Reply by antejavor** [reply]
Yes, this is true.
